modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2017-2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.sw360.workflow.generators;
+
+import org.eclipse.sw360.antenna.api.IAttachable;
+import org.eclipse.sw360.antenna.api.exceptions.AntennaException;
+import org.eclipse.sw360.antenna.model.artifact.Artifact;
+import org.eclipse.sw360.antenna.sw360.SW360MetaDataUpdater;
+import org.eclipse.sw360.antenna.sw360.rest.resource.components.SW360Component;
+import org.eclipse.sw360.antenna.sw360.rest.resource.releases.SW360Release;
+
+import java.io.IOException;
+import java.util.*;
+
+public class SW360UpdaterImpl {
+    private final String sw360AuthServerUrl;
+    private final String sw360User;
+    private final String sw360Password;
+    private final String sw360ClientId;
+    private final String sw360ClientPassword;
+    private final boolean sw360ProxyUse;
+    private final String sw360ProxyHost;
+    private final int sw360ProxyPort;
+    private final String sw360RestServerUrl;
+    private final String projectName;
+    private final String projectVersion;
+    private final boolean updateReleases;
+
+    public SW360UpdaterImpl(String sw360RestServerUrl, String sw360AuthServerUrl,
+                            String sw360User, String sw360Password, String sw360ClientId,
+                            String sw360ClientPassword, boolean sw360ProxyUse, String sw360ProxyHost,
+                            int sw360ProxyPort, String projectName, String projectVersion, boolean updateReleases) {
+        this.sw360RestServerUrl = sw360RestServerUrl;
+        this.sw360AuthServerUrl = sw360AuthServerUrl;
+        this.sw360User = sw360User;
+        this.sw360Password = sw360Password;
+        this.sw360ClientId = sw360ClientId;
+        this.sw360ClientPassword = sw360ClientPassword;
+        this.sw360ProxyUse = sw360ProxyUse;
+        this.sw360ProxyHost = sw360ProxyHost;
+        this.sw360ProxyPort = sw360ProxyPort;
+        this.projectName = projectName;
+        this.projectVersion = projectVersion;
+        this.updateReleases = updateReleases;
+    }
+
+    public Map<String, IAttachable> produce(Collection<Artifact> intermediates) throws AntennaException {
+        SW360MetaDataUpdater sw360MetaDataUpdater = new SW360MetaDataUpdater(
```

Comment:
If you intantiate the `SW360MetaDataUpdater` in the constructor, you might be able to get rid of most of the fields